tests/cupyx_tests/scipy_tests/signal_tests/test_filter_design.py

```diff
@@ -560,3 +567,125 @@ def test_w_or_N_types(self):
             w_out, h = signal.sosfreqz([1, 0, 0, 1, 0, 0], worN=w, fs=100)
             assert_array_almost_equal(w_out, [8])
             assert_array_almost_equal(h, [1])
+
+
+@testing.with_requires('scipy')
+class TestGroupDelay:
+    @testing.numpy_cupy_allclose(scipy_name='scp', rtol=1e-5, atol=1e-5)
+    def test_identity_filter(self, xp, scp):
+        w1, gd1 = scp.signal.group_delay((1, 1))
+        w2, gd2 = scp.signal.group_delay((1, 1), whole=True)
+        return w1, gd1, w2, gd2
+
+    @pytest.mark.skip(reason='firwin is not available on CuPy')
+    @testing.numpy_cupy_allclose(scipy_name='scp', rtol=1e-5, atol=1e-5)
+    def test_fir(self, xp, scp):
+        # Let's design linear phase FIR and check that the group delay
+        # is constant.
+        N = 100
+        b = scp.signal.firwin(N + 1, 0.1)
+        w, gd = scp.signal.group_delay((b, 1))
+        return w, gd
+
+    @testing.numpy_cupy_allclose(scipy_name='scp', rtol=1e-5, atol=1e-5)
+    def test_iir(self, xp, scp):
+        # Let's design Butterworth filter and test the group delay at
+        # some points against MATLAB answer.
+        b, a = scp.signal.butter(4, 0.1)
+        w = xp.linspace(0, xp.pi, num=10, endpoint=False)
+        w, gd = scp.signal.group_delay((b, a), w=w)
+        return w, gd
+
+    @testing.numpy_cupy_allclose(scipy_name='scp', rtol=1e-5, atol=1e-5)
+    def test_backward_compat(self, xp, scp):
+        # For backward compatibility, test if None act as a wrapper for default
+        w1, gd1 = scp.signal.group_delay((1, 1))
+        w2, gd2 = scp.signal.group_delay((1, 1), None)
+        return w1, gd1, w2, gd2
+
+    @testing.numpy_cupy_allclose(scipy_name='scp', rtol=1e-5, atol=1e-5)
+    def test_fs_param(self, xp, scp):
+        # Let's design Butterworth filter and test the group delay at
+        # some points against the normalized frequency answer.
+        b, a = scp.signal.butter(4, 4800, fs=96000)
+        w = xp.linspace(0, 96000 / 2, num=10, endpoint=False)
+        w, gd = scp.signal.group_delay((b, a), w=w, fs=96000)
+        return w, gd
+
+    @pytest.mark.parametrize(
+        'type_', [None, 'int8', 'int16', 'int32', 'int64'])
+    @testing.numpy_cupy_allclose(scipy_name='scp', rtol=1e-5, atol=1e-5)
+    def test_N_types(self, type_, xp, scp):
+        # Measure at 8 equally-spaced points
+        N = 8
+        if type_ is not None:
+            wrapper = getattr(xp, type_)
+            N = wrapper(N)
+
+        w, gd = scp.signal.group_delay((1, 1), N)
+        return w, gd
+
+    @pytest.mark.parametrize('w', [8.0, 8.0+0j])
+    @testing.numpy_cupy_allclose(scipy_name='scp', rtol=1e-5, atol=1e-5)
+    def test_w_types(self, w, xp, scp):
```

Comment:
Older scipy seems to return a result with a different dtype.
https://ci.preferred.jp/cupy.linux.cuda102/135386/ (scipy==1.6.\*)
https://ci.preferred.jp/cupy.linux.cuda112/135394/ (scipy==1.7.\*)
```suggestion
    @testing.with_requires('scipy>=1.8')
    def test_w_types(self, w, xp, scp):
```